### PR TITLE
Lightning installation instructions improvements

### PR DESCRIPTION
**Pull request by isaacdevlugt**
**Context:** Our installation instructions are unclear for users.

**Description of the Change:** Point to pennylane.ai/install where possible.

**Benefits:** Better installation instructions and more clarity for users.

**Possible Drawbacks:** Hopefully none.

**Related GitHub Issues:**

### CODE REVIEW

**Comment by mlxd** [comment]
Hey Isaac, we should wait until #832 is merged before tackling anything here, as the instructions may differ again.

**Comment by isaacdevlugt** [comment]
> Hey Isaac, we should wait until #832 is merged before tackling anything here, as the instructions may differ again.

If we're doing this just to avoid merge conflicts, all good! Just want to make sure though that the scope of this PR doesn't cover source installation for anything 👍. All good on my side to wait for #832

File: README.rst
Comment (trbromley) [inline README.rst:1]: Instead of having installation instructions replicated in the `README` _and_ docs, perhaps we could just make the `README` instructions very briefly redirect to the [docs installation](https://xanaduai-pennylane--848.com.readthedocs.build/projects/lightning/en/848/dev/installation.html)

File: README.rst
Comment (josh146) [inline README.rst:87]:
```suggestion
.. note::

    The below contains instructions for installing Lightning Qubit ***from source***. For most cases, *this is not required* and one can simply use the `installation instructions at pennylane.ai/install <https://pennylane.ai/install>`_.
    If those instructions do not work for you, or you have a more complex build environment that requires building from source, then consider reading on.
```

File: README.rst
Comment (josh146) [inline README.rst:220]:
```suggestion
    If those instructions do not work for you, or you have a more complex build environment that requires building from source, then consider reading on.
```

File: README.rst
Comment (josh146) [inline README.rst:298]:
```suggestion
.. note::

    The below contains instructions for installing Lightning-Kokkos ***from source***. For most cases, one can install Lightning-Kokkos via Spack or Docker by the `installation instructions at pennylane.ai/install <https://pennylane.ai/install/#high-performance-computing-and-gpus>`_.
    If those instructions do not work for you, or you have a more complex build environment that requires building from source, then consider reading on.
```

File: README.rst
Comment (josh146) [inline README.rst:291]: with lightning.kokkos, I believe there may be more cases than qubit/gpu where you need to install from source (e.g., if you want to use GPUs)

**Review by josh146 — APPROVED** [review]
Thanks @isaacdevlugt! Good from my end, although will defer to the perf team for technical review.

File: README.rst
Comment (isaacdevlugt) [inline README.rst:87]:
```suggestion
    The below contains instructions for installing Lightning-Qubit ***from source***. For most cases, *this is not required* and one can simply use the `installation instructions at pennylane.ai/install <https://pennylane.ai/install>`_.
```

File: README.rst
Comment (isaacdevlugt) [reply README.rst:292]: Thanks! Does this sentence need to be modified 🤔?

File: README.rst
Comment (isaacdevlugt) [reply README.rst:1]: I think the README instructions provide what's in the installation page (e.g., https://github.com/PennyLaneAI/pennylane-lightning/blob/e32f68c0f3d370c676ae361f1d78a8305db97770/doc/lightning_qubit/installation.rst#L1)

**Comment by isaacdevlugt** [comment]
Will be moving these changes to https://github.com/PennyLaneAI/pennylane-lightning/pull/851

**Pull request closed without merging**
